### PR TITLE
Changes Related to https://github.com/ldtteam/Structurize/pull/406

### PR DESCRIPTION
**Pull request by Justen-N**
DO NOT MERGE UNLESS ldtteam/Structurize#406 is accepted. 
# Changes proposed in this pull request:
- Abstract BlockHut Now implements  IBlockHut, a new marker interface. 
-
-

Review please

### CODE REVIEW

File: src/api/java/com/minecolonies/api/blocks/AbstractBlockHut.java [hunk 1]
```diff
@@ -77,7 +77,7 @@
     private static final VoxelShape SHAPE = VoxelShapes.create(0.1, 0.1, 0.1, 0.9, 0.9, 0.9);
 
     /**
-     * The hut's lower-case building-registry-compatible name.
+     * The hut's lower-case building-registry-compatible name.s
```
Comment (ravenbuilder934) [inline src/api/java/com/minecolonies/api/blocks/AbstractBlockHut.java]: is this change intentional?

Comment (Justen-N) [reply src/api/java/com/minecolonies/api/blocks/AbstractBlockHut.java]: No. Also, damn you're fast.

Comment (ravenbuilder934) [reply src/api/java/com/minecolonies/api/blocks/AbstractBlockHut.java]: heh. fastest triage in the west, and all that

Comment (Justen-N) [reply src/api/java/com/minecolonies/api/blocks/AbstractBlockHut.java]: In fact, I didn't even touch that line, my fork just has weird crap happening. I no longer have direct access to the repo, so that's a thing.

File: src/api/java/com/minecolonies/api/blocks/AbstractBlockHut.java [hunk 2]
```diff
@@ -49,7 +49,7 @@
  * All AbstractBlockHut[something] should extend this class.
  */
 @SuppressWarnings("PMD.ExcessiveImports")
-public abstract class AbstractBlockHut<B extends AbstractBlockHut<B>> extends AbstractBlockMinecolonies<B> implements IBuilderUndestroyable, IAnchorBlock
+public abstract class AbstractBlockHut<B extends AbstractBlockHut<B>> extends AbstractBlockMinecolonies<B> implements IBuilderUndestroyable, IAnchorBlock, IBlockHut
```
Comment (Raycoms) [inline src/api/java/com/minecolonies/api/blocks/AbstractBlockHut.java:52]: Can't we move IHutBlock to MineColonies, and in MineColonies extend structurename and use our own version which does the hut check or so?

**Comment by Justen-N** [comment]
By the way, when this is reviewed, I would like to ask: Do we want to do away with the blockhut naming convention for blockhuts? Currently, this would require an overhaul of multiple sections of code and translation, and I don't want to start in on it if we don't want to do it

**Comment by ravenbuilder934** [comment]
my two cents: get rid of it, but in 1.17, so we don't need to worry about datafixeruppers/etc

File: src/api/java/com/minecolonies/api/blocks/AbstractBlockHut.java (hunk 2 above)
Comment (Raycoms) [reply src/api/java/com/minecolonies/api/blocks/AbstractBlockHut.java:52]: If we go as far as Improving this, then we should remove structurize having to know about huts

Comment (Justen-N) [reply src/api/java/com/minecolonies/api/blocks/AbstractBlockHut.java:52]: If I were to write this on MC's side, I would just check for membership in the AbstractHutBlock class, and not worry about the interface at all.  However, this check only matters for schematics, which seems to be why we would want it in Structurize.

Comment (Raycoms) [reply src/api/java/com/minecolonies/api/blocks/AbstractBlockHut.java:52]: Well, it's only for schematics, but who defines special treatment of schematics should be the mInecolonies side, and not the structurize side

Comment (Justen-N) [reply src/api/java/com/minecolonies/api/blocks/AbstractBlockHut.java:52]: Alright. I'll see what I can on that front

Comment (Justen-N) [reply src/api/java/com/minecolonies/api/blocks/AbstractBlockHut.java:52]: Hold on. I'm gonna cancel this PR and we just need to convince Orion to wait on these changes. Section/style/level should all be in the Metadata changes. This is adding a 'fix' for something that really will be fixed at a later date in a better way.

Comment (uecasm) [reply src/api/java/com/minecolonies/api/blocks/AbstractBlockHut.java:52]: It would be nice if the levelled decorations could have a level selector similar to huts (would save a lot of "why is it building level 1 instead of level 3" questions).  Having said that, that's tricky with the current UI arrangement and you probably wouldn't want to treated the levelled decos as "huts" as far as appearing individually in the first selector, so it's not as easy as that sounds.
